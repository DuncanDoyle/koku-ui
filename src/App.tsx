import React from 'react';
import { hot } from 'react-hot-loader';
import { I18nextProvider } from 'react-i18next';
import { Route, Switch } from 'react-router-dom';
import { initI18next } from './i18next';
import { styled } from './styles/styled';
import asyncComponent from './utils/asyncComponent';

const NotFound = asyncComponent(() =>
  import(/* webpackChunkName: "notFound" */ './components/notFound')
);

const Home = asyncComponent(() =>
  import(/* webpackChunkName: "home" */ './components/home')
);

interface Props {
  language: string;
}

interface State {
  isLoaded: boolean;
}

const Header = styled('header');
const Content = styled('main');
const Navigation = styled('aside');

class App extends React.Component<Props, State> {
  public state = {
    isLoaded: false
  };

  private i18n = initI18next(this.props.language);

  public render() {
    return (
      <I18nextProvider i18n={this.i18n}>
        <>
          <Header>Header</Header>
          <Navigation>Side Navigation</Navigation>
          <Content>
            <Switch>
              <Route exact component={Home} path="/" />
              <Route component={NotFound} />
            </Switch>
          </Content>
        </>
      </I18nextProvider>
    );
  }
}

export default hot(module)(App);
