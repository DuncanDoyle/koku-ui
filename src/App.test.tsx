jest.mock('./i18next');

import { shallow } from 'enzyme';
import React from 'react';
import App from './App';

// TODO: Replace with useful test
test('renders', () => {
  const view = shallow(<App language="en" />);
  expect(view).toMatchSnapshot();
});
