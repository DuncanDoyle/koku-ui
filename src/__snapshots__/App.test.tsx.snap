// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<I18nextProvider
  i18n={Object {}}
>
  <React.Fragment>
    <glamorous(header)>
      Header
    </glamorous(header)>
    <glamorous(aside)>
      Side Navigation
    </glamorous(aside)>
    <glamorous(main)>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </glamorous(main)>
  </React.Fragment>
</I18nextProvider>
`;
